### PR TITLE
Add mass download support for related documents

### PR DESCRIPTION
**Pull request by mlorencik**
Przechwytywanie

### CODE REVIEW

File: modules/Documents/models/Record.php [hunk 1]
```diff
@@ -110,6 +109,50 @@ public function downloadFile()
...
+	/**
+	 * Download files.
+	 *
+	 * @param int[] $recordsIds
+	 */
+	public static function downloadFiles($recordsIds)
+	{
+		$zip = new ZipArchive();
+		$postfix = time() . '_' . random_int(0, 1000);
+		$zipPath = 'storage/';
```
Comment (TomaszQr) [inline modules/Documents/models/Record.php]: not storage -> cache

File: public_html/layouts/basic/modules/Vtiger/resources/RelatedList.js [hunk 2]
```diff
@@ -1123,12 +1118,183 @@ jQuery.Class("Vtiger_RelatedList_Js", {
...
+				}
+			}
+			self.checkSelectAll();
+			self.writeSelectedIds(selectedIds);
+			self.writeExcludedIds(excludedIds);
+		});
+	},
+	registerMainCheckBoxClickEvent: function() {
+		const self = this;
+		this.getRelatedContainer().on("click", "#listViewEntriesMainCheckBox", function() {
```
Comment (TomaszQr) [inline public_html/layouts/basic/modules/Vtiger/resources/RelatedList.js]: " => ' everywhere

File: layouts/basic/modules/Vtiger/RelatedListContents.tpl [hunk 3]
```diff
@@ -79,13 +97,20 @@
...
+						data-recordUrl="{$RELATED_RECORD->getDetailViewUrl()}"
+					{/if}>
 					{assign var=COUNT value=0}
 					<td class="{$WIDTHTYPE} noWrap leftRecordActions"
 						{if $RECORD_COLORS['leftBorder']}style="border-left-color: {$RECORD_COLORS['leftBorder']};"{/if}>
+						{if isset($RELATED_LIST_LINKS['RELATEDLIST_MASSACTIONS'])}
+							<div>
+								<input type="checkbox" value="{$RELATED_RECORD->getId()}"
+								title="{\App\Language::translate('LBL_SELECT_SINGLE_ROW')}"
+								class="listViewEntriesCheckBox"/>
```
Comment (TomaszQr) [inline layouts/basic/modules/Vtiger/RelatedListContents.tpl]: relatedListViewEntriesCheckBox

File: modules/Documents/models/Record.php [hunk 4]
```diff
@@ -110,6 +109,50 @@ public function downloadFile()
...
+					if (file_exists($filePath . $savedFile)) {
+						$zip->addFile($filePath . $savedFile, basename($documentModel->get('filename')));
+						$documentModel->updateDownloadCount();
+					}
+				}
+			}
+		}
+		$zip->close();
+		header('content-type: ' . \App\Fields\File::getMimeContentType($fileName));
+		header('Content-Disposition: attachment; filename="' . basename($fileName) . '";');
```
Comment (rskrzypczak) [inline modules/Documents/models/Record.php]:
```suggestion
		header('content-disposition: attachment; filename="' . basename($fileName) . '";');
```

File: modules/Vtiger/actions/RelationAjax.php [hunk 5]
```diff
@@ -507,4 +462,22 @@ public function calculate(App\Request $request)
...
+	public function massDownload(App\Request $request)
+	{
+		$relatedModuleName = $request->getByType('relatedModule', 2);
+		$records = $this->getRecordsListFromRequest($request);
+		if (1 === count($records)) {
+			$documentRecordModel = Vtiger_Record_Model::getInstanceById($records[0], $relatedModuleName);
+			$documentRecordModel->downloadFile();
+			$documentRecordModel->updateDownloadCount();
+		} else {
+			Documents_Record_Model::downloadFiles($records);
```
Comment (rskrzypczak) [inline modules/Vtiger/actions/RelationAjax.php]: ?

File: modules/Vtiger/models/RelationListView.php [hunk 6]
```diff
@@ -425,21 +427,31 @@ public function getLinks()
...
-		]);
-		$relatedLink['RELATEDLIST_MASSACTIONS_ADV'][] = Vtiger_Link_Model::getInstanceFromValues([
-			'linktype' => 'RELATEDLIST_MASSACTIONS_ADV',
-			'linklabel' => 'LBL_QUICK_EXPORT_TO_EXCEL',
-			'linkurl' => "javascript:Vtiger_RelatedList_Js.triggerMassAction('index.php?module=Campaigns&action=RelationAjax&mode=exportToExcel&src_record={$id}&relatedModule={$relatedModuleName}','sendByForm')",
-			'linkclass' => '',
-			'linkicon' => 'fas fa-file-excel',
-		]);
-
+		if ('Campaigns' === $parentRecordModuleName) {
```
Comment (rskrzypczak) [inline modules/Vtiger/models/RelationListView.php]: ?

File: modules/Vtiger/actions/RelationAjax.php (hunk 5 above)
Comment (mlorencik) [reply modules/Vtiger/actions/RelationAjax.php]: one file or files zip

File: modules/Vtiger/models/RelationListView.php (hunk 6 above)
Comment (mlorencik) [reply modules/Vtiger/models/RelationListView.php]: earlier only campaigns support massactions

**Pull request closed without merging**